### ISSUE TITLE
OpenType support missing for "<FONT_NAME>", script <NUMBER>

### ISSUE THREAD

**Issue by RandallFlagg** [telegramdesktop/tdesktop#28083]
### Steps to reproduce

Download the latest binary for linux from github release page.
I  downloaded:
https://github.com/telegramdesktop/tdesktop/releases/download/v5.1.7/tsetup.5.1.7.tar.xz

When I tried to get the files from the offical site I got the same package with offcourse the same results.

### Expected behaviour

Non english fonts will look good

### Actual behaviour

Non english fonts look very spacy and everything is difficult to read

### Operating system

Void Linux 6.6.35

### Version of Telegram Desktop

5.1.7

### Installation source

Static binary from official website

### Logs

```shell
This shows in the terminal when I start. Otherwise I don't see any errors except of the visuals I reported before. If there are logs from somewhere else please direct me to them.

QPainter::begin: Paint device returned engine == 0, type: 2
QWidget::render: Cannot render with an inactive painter
qt.gui.imageio.jpeg: ./lib/jpegli/decode_scan.cc:535: Incomplete scan detected.
qt.gui.imageio.jpeg: ./lib/jpegli/decode_scan.cc:535: Incomplete scan detected.
  OpenType support missing for "Open Sans", script 20
  OpenType support missing for "vazirmatn ui nl", script 20
```

**Comment by ilya-fedin** [comment]
I can't reproduce, provide a screenshot of the issue

**Comment by RandallFlagg** [comment]
![TGBUG2](https://github.com/telegramdesktop/tdesktop/assets/434681/94122fdb-e75f-4d80-bb60-29d0de964ffc)
![TGBUG](https://github.com/telegramdesktop/tdesktop/assets/434681/bf82f19d-dec7-465d-998b-a5ca237e79ba)

**Comment by ilya-fedin** [comment]
Sorry but I don't see anything wrong on the screenshot. Looks like there's no bug.

**Comment by RandallFlagg** [comment]
The letters are too seperated. They are suppose to be closer. See the space difference between the hebrew and the english.
You can try and see how it looks in Windows if you have.  I will also try to bring a reference.

**Comment by ilya-fedin** [comment]
Sorry but I don't see a bug here. You may not like how it looks but there's no bug.

**Comment by RandallFlagg** [comment]
This is not only "not looking good" but a "not usable app" issue.

The words are very hard to read that I am using the web version in the meantime. You don't see the difference compared to Windows?
![TGBUG3](https://github.com/telegramdesktop/tdesktop/assets/434681/81f43f44-9e40-453b-8862-de8ed5aab1d1)

Maybe the bug is not what I wrote, but the reason I wrote it was because it has to do with fonts and that is what the message is saying.

If this is not a bug, can you point me to a documentation of write here how to fix it so I will be able to use the desktop version in Linux?

**Comment by ilya-fedin** [comment]
Your Windows screenshot looks worse to me regarding Hebrew font (too bold). But english looks the same to me.

**Comment by ilya-fedin** [comment]
> If this is not a bug, can you point me to a documentation or write here how to fix it so I will be able to use the desktop version in Linux?

I'm not sure which documentation you're talking about. tdesktop has no documentation. I also don't see anything to fix, sorry.

**Comment by RandallFlagg** [comment]
You don't see the difference between the 2 images I shared?

**Comment by ilya-fedin** [comment]
I only see that Hebrew font is bolder on Windows. Apparently because it has a different one than you on Linux. That's all. I don't see any bug.

**Comment by ilya-fedin** [comment]
I recommend you to try user-friendly distros like Ubuntu - maybe their font rendering would be better for you.

**Comment by RandallFlagg** [comment]
I don't want to use Ubuntu and the differnce is not only a bolder font. The problem is a very bad font. Maybe a default one. I don't know. The problem is that the letters are very spaced. VERY!

How do I change the font of Telegram? What font does it need? Do I need to install a font? Where should I put it?

**Comment by ilya-fedin** [comment]
> How do I change the font of Telegram?

Settings -> Chat settings -> Font family

> What font does it need?

tdesktop has embedded Open Sans font. Anything not covered by it should be covered by your system fonts in a platform-defined manner defined by third party libraries like Qt, fontconfig, freetype and harfbuzz.

**Comment by RandallFlagg** [comment]
Thanks. Going to try. Will update later.

**Comment by RandallFlagg** [comment]
The problem is indeed to do with the fonts. I changed to "Liberation Sans".
I checked it with version 4.16.8 and that works great. There is a change in the font system of Telegram which was introduced in  a later version. As I mentioned I am using 5.1.7 and that caused the problem.

**Comment by ilya-fedin** [comment]
well, things do change

**Comment by RandallFlagg** [comment]
I don't think that reverting is the solution but to understand what changed in the default behaviour and fix it in version 5.

**Comment by ilya-fedin** [comment]
Well, I don't even see a bug...

**Comment by Tynach** [comment]
> Well, I don't even see a bug...

@ilya-fedin, I think I understand what to at least look at when looking for the bug. Certain languages are particularly sensitive to the spacing of characters, because characters are meant to flow from one to another.

One more extreme example is Bengali, which has a bar near the top of the text that is meant to be continuously connected. In this case, a break in that line has actual meaning, and improper spacing due to a bad (or badly rendered) font can make the same text that used to look fine, become gibberish. To be fair, I only know about this because of reading something somewhere once about a language that had a bar go through words continuously like that, and I had to look up the name of the language before posting.

Anyway, when I look at those screenshots, especially the not-bolded text at the bottom, I see a LOT of spaces between pretty much every single letter. I know nothing about Hebrew, but something tells me that *that* much spacing between every single letter isn't normal. A few letters look like they have normal spacing, but not most of them.

If I had to guess, the change that fixed the appearance of capital 'I's (something I think is a *great* thing, by the way, and I definitely don't want to revert that *at all*) accidentally set in stone what language all text is being interpreted as by Harfbuzz, and so it's not reading Hebrew-specific font rules from Hebrew fonts, and thus might not be applying kerning rules and the like.

In other words, this **definitely** looks like a bug to me.

... Or it would if I could reproduce it. I'm on Linux too, and using the default font everything looks fine, I don't see abnormal spacing. Perhaps 5.2 fixed the issue, @RandallFlagg?

**Comment by ilya-fedin** [comment]
I think it's just the person has some bad font installed and some Qt update changed the fallback

**Comment by Tynach** [comment]
> I think it's just the person has some bad font installed and some Qt update changed the fallback

Very possible. Either way, for me the fallback currently works fine. I even threw some Bengali text at it, and it appeared (to me, someone who does not know how to read it at all) fine as well. At the very least, the line was only broken when it was supposed to be.
